### PR TITLE
[2] Tabbed process list draft (unstyled)

### PR DESCRIPTION
**Pull request by JulianWielga**

### CODE REVIEW

File: ui/client/components/AddProcessDialog.tsx [hunk 1]
```diff
@@ -1,62 +1,72 @@
...
+    showProcess(processId)
   }
 
   render() {
-    const titleStyles = EspModalStyles.headerStyles("#2d8e54", "white")
-    const nameValidators = prepareNameValidators(this.props.clashedNames)
+    const {message, clashedNames, categories, isOpen} = this.props
+    const {processId} = this.state
+
+    const titleStyles = EspModalStyles.headerStyles("#2D8E54", "white")
```
Comment (lciolecki) [inline ui/client/components/AddProcessDialog.tsx:64]: Please change `EspModalStyles` to `NkModalStyles`.

File: ui/client/components/table/DeployedFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import TableSelect from "./TableSelect"
```
Comment (lciolecki) [inline ui/client/components/table/DeployedFilter.tsx:1]: Please change `DeployedFilter` to `StatusFilter`. Right now we filter only by deployed / not deployed - but in future who knows ;)

File: ui/client/components/table/SearchFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function SearchFilter(props: FilterProps) {
+  const {value, onChange} = props
+  const fillIconClass = isEmpty(value) ? "search-icon-fill" : "search-icon-fill-filter"
+  return (
+    <div id="table-filter" className="input-group">
+      <div className="search-container">
+        <input
+          type="text"
+          placeholder="Filter by text..."
```
Comment (lciolecki) [inline ui/client/components/table/SearchFilter.tsx:14]: Missing text translation

File: ui/client/components/table/SubprocessFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import {Option, FilterProps} from "./FilterTypes"
```
Comment (lciolecki) [inline ui/client/components/table/SubprocessFilter.tsx:1]: Please change `SubprocessFilter` to `ProcessTypeFilter`.

File: ui/client/components/table/TableSelect.tsx [hunk 5]
```diff
@@ -2,7 +2,16 @@ import React from "react"
...
+type Props = {
+  defaultValue: $TodoType,
+  onChange: $TodoType,
+  options: $TodoType[],
+  isMulti: boolean,
+  isSearchable: boolean,
+  placeholder: string,
+}
+
+export default class TableSelect extends React.Component<Props> {
```
Comment (lciolecki) [inline ui/client/components/table/TableSelect.tsx:14]: `React.Component` need to params: Props and State. IDEA shows me error with missing State..

File: ui/client/containers/AdminPage.js [hunk 6]
```diff
@@ -4,17 +4,18 @@ import {withRouter} from "react-router-dom"
...
 import {nkPath} from "../config"
-import CustomProcesses from "./admin/CustomProcesses"
+import {CustomProcesses, CustomProcessesTabData} from "./admin/CustomProcesses"
 import SearchComponents from "./admin/SearchComponents"
 import Services from "./admin/Services"
 import UnusedComponents from "./admin/UnusedComponents"
+import {Page} from "./Page"
 
 export class AdminPage extends React.Component {
   tabs = [
```
Comment (lciolecki) [inline ui/client/containers/AdminPage.js:14]: Why all components don't have `TabData`? You have changed it only at CustomProcesses, I wonder why?

File: ui/client/containers/EnsureArray.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import {isArray} from "lodash"
```
Comment (lciolecki) [inline ui/client/containers/EnsureArray.tsx:1]: I wonder.. sometimes we use camelcase start with big letter anothertime we use it by start small letter? Can you tell my why? Can we assume that when we export only methods then we start with small letter?  Alsow I think this util shoul be in package like utils.

It's only array tools -> it should named `arrayUtils`.

File: ui/client/containers/EnsureArray.tsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+export function ensureArray<T extends any>(maybeArray: T | T[] = []): T[] {
```
Comment (lciolecki) [inline ui/client/containers/EnsureArray.tsx:3]: `convertToArray`

File: ui/client/containers/ProcessTabs.styl [hunk 9]
```diff
@@ -0,0 +1,35 @@
+:local
```
Comment (lciolecki) [inline ui/client/containers/ProcessTabs.styl:1]: Formatting?

File: ui/client/containers/ProcessTabs.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import {TabLink} from "./TabLink"
+import {TabRoute} from "./TabRoute"
+
+type TabData = {path: string, header: string, Component: ComponentType}
+
+function Tabs({tabs}: {tabs: TabData[]}) {
+  return (
+    <div className={styles.tabsWrap}>
+      <div className={styles.tabs}>
+        {tabs.map(r => <TabLink key={r.path} {...r}/>)}
```
Comment (lciolecki) [inline ui/client/containers/ProcessTabs.tsx:15]: `r`? place replace it by `tab`.

File: ui/client/containers/TabContent.tsx [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import React, {PropsWithChildren} from "react"
```
Comment (lciolecki) [inline ui/client/containers/TabContent.tsx:1]: Can we create module `Tab` and move all tab components there?

File: ui/client/containers/hooks/useSearchQuery.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import {defaults} from "lodash"
+import * as queryString from "query-string"
```
Comment (lciolecki) [inline ui/client/containers/hooks/useSearchQuery.tsx:2]: You used two times import..

File: ui/client/containers/processesTable/ProcessesTable.tsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  const onPageChange = useCallback(page => setQuery({...query, page}), [query])
+  const onSort = useCallback(({column, direction}) => setQuery({...query, column, direction}), [query])
+
+  return (
+    <>
+      <LoaderSpinner show={isLoading}/>
+      <Table
+        {...passProps}
+        className={"esp-table"}
+        noDataText={isLoading ? "Loading data..." : "No matching records found."}
```
Comment (lciolecki) [inline ui/client/containers/processesTable/ProcessesTable.tsx:41]: Missing text translate

File: ui/client/stylesheets/animations.styl [hunk 14]
```diff
@@ -0,0 +1,33 @@
+:local
```
Comment (lciolecki) [inline ui/client/stylesheets/animations.styl:1]: Formatting.

File: ui/client/components/AddProcessDialog.tsx (hunk 1 above)
Comment (JulianWielga) [reply ui/client/components/AddProcessDialog.tsx:64]: fixed in [6]

File: ui/client/components/table/DeployedFilter.tsx (hunk 2 above)
Comment (JulianWielga) [reply ui/client/components/table/DeployedFilter.tsx:1]: [6]

File: ui/client/components/table/TableSelect.tsx (hunk 5 above)
Comment (JulianWielga) [reply ui/client/components/table/TableSelect.tsx:14]: no, `interface Component<P = {}, S = {}, SS = any> extends ComponentLifecycle<P, S, SS> { }`,  check your config.

btw changed to FC later

File: ui/client/containers/EnsureArray.tsx (hunk 7 above)
Comment (JulianWielga) [reply ui/client/containers/EnsureArray.tsx:1]: changed to arrayUtils

File: ui/client/containers/ProcessTabs.styl (hunk 9 above)
Comment (JulianWielga) [reply ui/client/containers/ProcessTabs.styl:1]: its stylus, "indents'" lives matters

File: ui/client/containers/ProcessTabs.tsx (hunk 10 above)
Comment (JulianWielga) [reply ui/client/containers/ProcessTabs.tsx:15]: was from "route", replaced with "data"

File: ui/client/containers/hooks/useSearchQuery.tsx (hunk 12 above)
Comment (JulianWielga) [reply ui/client/containers/hooks/useSearchQuery.tsx:2]: has to be this way, one for commonjs module, one for type

File: ui/client/containers/AdminPage.js (hunk 6 above)
Comment (JulianWielga) [reply ui/client/containers/AdminPage.js:14]: i don't like pattern from "UnusedComponents", makes problems with ts. I changed only what I had to

**Pull request merged**